subprojects/docs/src/docs/userguide/lazyConfiguration.adoc

```diff
@@ -15,40 +15,226 @@
 [[lazy_configuration]]
 == Lazy Configuration
 
+As a build grows in complexity, knowing when and where a particular value is configured can become difficult to reason about. Gradle provides several ways to manage this complexity with lazy configuration.
 
 [[sec:lazy_properties]]
 === Lazy properties
 
 [NOTE]
 ====
 
-The provider API is currently <<feature_lifecycle,incubating>>. Please be aware that the DSL and other configuration may change in later Gradle versions.
+The Provider API is currently <<feature_lifecycle,incubating>>. Please be aware that the DSL and other configuration may change in later Gradle versions.
 
 ====
 
-[[sec:creating_new_property_instances]]
-=== Creating new instances
+First, Gradle provides lazy properties, which delay the calculation of a property’s value until it’s absolutely required. This provides two main benefits to build script and plugin authors:
```

Comment:
nit-pick: By saying "First,..." you're implying that you're about to compare it to something or enumerate several things (i.e. "Second,...", "Third,..."), but you don't really do that.  I'd just remove "First," from this line.

Reply:
Good point, I removed it.